### PR TITLE
Resposne object for Thread Support

### PR DESCRIPTION
**Pull request by steven-foster**
Hi,
This was my idea to allow the user to send a message from the slack plugin and then add a thread to the message. I added a response object with a few properties that will be accessible inside the pipeline.
example:
```
response = slackSend botUser: true, channel: 'general', message: 'Hello'
slackSend botUser: true, channel: "${response.threadId}", message: 'this is a thread'
```

### CODE REVIEW

File: src/main/java/jenkins/plugins/slack/workflow/SlackSendStep.java [hunk 1]
```diff
@@ -197,7 +197,7 @@ public FormValidation doCheckToken(@QueryParameter String value) {
         transient TaskListener listener;
 
         @Override
-        protected Void run() throws Exception {
+        protected SlackResponse run() throws Exception {
```
Comment (iulianstana) [inline src/main/java/jenkins/plugins/slack/workflow/SlackSendStep.java]: I'm not sure of all dependencies of this change....

File: src/main/java/jenkins/plugins/slack/workflow/SlackSendStep.java [hunk 2]
```diff
@@ -257,12 +257,19 @@ protected Void run() throws Exception {
             }else{
                 publishSuccess = slackService.publish(step.message, color);
             }
-            if (!publishSuccess && step.failOnError) {
+            SlackResponse response = null;
+            if (publishSuccess) {
```
Comment (iulianstana) [inline src/main/java/jenkins/plugins/slack/workflow/SlackSendStep.java]: Can you write a test for this?

**Review by iulianstana — COMMENTED** [review]
I am not a Java developer ... I am afraid that some of this changes will not get a proper review. 

Can you tell me the exact scenario of your changes? 

Where excatly is `"slackSend botUser: true, channel: "${response.threadId}", message: 'this is a thread'"` placed?

**Comment by mcinteer** [comment]
I would find these changes in conjunction with your other [PR](https://github.com/jenkinsci/slack-plugin/pull/377) @iulianstana very useful. @steven-foster - any chance you coul address the comments here?

**Comment by steven-foster** [comment]
I will try to make a proper attempt at this after the original PR on the jenkinsci repo is merged, this was more of a mockup to show the idea :)

**Comment by ben-gineer** [comment]
It would be great to see this pulled in.

**Pull request closed without merging**